File: strings/caesarCipherEncryptor/caesarCipherEncryptor.py
# O(n) time | O(n) space


def caesarCipherEncryptor(string, key):
    newLetters = []
    newKey = key % 26
    alphabet = list("abcdefghijklmnopqrstuvwxyz")

    for letter in string:
        newLetters.append(getNewLetter(letter, newKey, alphabet))

    return "".join(newLetters)


def getNewLetter(letter, key, alphabet):
    newIndex = (alphabet.index(letter) + key) % 26
    return alphabet[newIndex]


# Example usage:
if __name__ == "__main__":
    input_string = "xyz"
    input_key = 2
    encrypted_string = caesarCipherEncryptor(input_string, input_key)
    print(f"Encrypted string: {encrypted_string}")  # Output: "zab"
    input_string = "abc"
    input_key = 3
    encrypted_string = caesarCipherEncryptor(input_string, input_key)
    print(f"Encrypted string: {encrypted_string}")  # Output: "def"
    input_string = "hello"
    input_key = 5
    encrypted_string = caesarCipherEncryptor(input_string, input_key)
    print(f"Encrypted string: {encrypted_string}")  # Output: "mjqqt"
    input_string = "xyz"
    input_key = 26
    encrypted_string = caesarCipherEncryptor(input_string, input_key)
    print(f"Encrypted string: {encrypted_string}")  # Output: "xyz"

File: strings/caesarCipherEncryptor.js
// O(n) time | O(n) space

function caesarCipherEncryptor(string, key) {
  const alphabet = 'abcdefghijklmnopqrstuvwxyz'
  let resultArr = []

  for(let i = 0; i < string.length; i++) {
    let indexOfLetter = alphabet.indexOf(string[i])
    let encriptedIndex = indexOfLetter + key

    if(encriptedIndex >= alphabet.length) {
         encriptedIndex = encriptedIndex % alphabet.length 
    } 
    
    resultArr.push(alphabet[encriptedIndex])
  }
  return resultArr.join('')
}

// Example usage:
console.log(caesarCipherEncryptor('xyz', 2)); // Output: 'zab'
console.log(caesarCipherEncryptor('abc', 3)); // Output: 'def'
console.log(caesarCipherEncryptor('hello', 5)); // Output: 'mjqqt'
console.log(caesarCipherEncryptor('xyz', 28)); // Output: 'zab' (28 % 26 = 2